pkg/config/template/filebased.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package template
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/spf13/afero"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	_ Template = (*FileBasedTemplate)(nil)
+)
+
+// FileBasedTemplate is a JSON Template stored in a file - when it's Template.Content is accessed, that file is read.
+// This is the usual type of Template monaco uses.
+type FileBasedTemplate struct {
+	fs   afero.Fs
+	path string
+}
+
+func (t *FileBasedTemplate) Id() string {
+	return t.path
+}
+
+func (t *FileBasedTemplate) Content() (string, error) {
+	b, err := afero.ReadFile(t.fs, t.path)
```

Comment:
What do you think about ignoring an error?

Since the existence of the file is already checked in the constuction method, the error can be ignored. If it occurs, just write ERROR message and go to panic mode.

For more safety, performing a test read during creation is possible. And if it is necessary for performance, it can be done in parallel.

Reply:
I consider it at the start but I would not.
While it generally won't happen, it can still happen that a file gets deleted or becomes inaccessible during deployment. Consider that even with parallel deployments, there might be an hour or two between a template being initially created and then actually read and rendered for an API call.

While unlikely, I don't think we should just ignore the error, so I've added it to the interface and handle it as an error everywhere.

Reply:
As you said, it is rarely possible. In my opinion, the assumption that touching configuration is forbidden once the process starts is quite common. So panicking in that case is acceptable.

It is more of a discussion about possibilities. The current solution is acceptable to me.

Reply:
Let's discuss this in the team some time soon!

IMO panicking does not fit our tendency to handle problems and continue (e.g. if one deployment tree fails, we continue others). Would be worth discussing and defining that as a shared design philosophy and with that ban must uses of panic.
This is unlikely to happen, but it can - clean error handling is IMO a small cost to pay for not breaking a full deployment because some file is corrupt or was removed.